modules/Core/src/main/java/org/terasology/core/world/generator/rasterizers/TreeRasterizer.java

```diff
@@ -15,108 +15,57 @@
-                .put('A', new LSystemRule("[&FFBFA]////[&BFFFA]////[&FBFFAFFA]", 1.0f))
-                .put('B', new LSystemRule("[&FFFAFFFF]////[&FFFAFFF]////[&FFFAFFAA]", 0.8f)).build();
-        TreeGenerator oakVariationTree = new TreeGeneratorLSystem("FFFFFFA", rules, 4, (float) Math.toRadians(35)).setLeafType(blockManager.getBlock("core:GreenLeaf"))
-                .setBarkType(blockManager.getBlock("core:OakTrunk")).setGenerationProbability(0.08f);
-
-        // A red tree
-        rules = ImmutableMap.<Character, LSystemRule>builder()
-                .put('A', new LSystemRule("[&FFAFF]////[&FFAFF]////[&FFAFF]", 1.0f)).build();
-        TreeGenerator redTree = new TreeGeneratorLSystem("FFFFFAFAFAF", rules, 4, (float) Math.toRadians(40)).setLeafType(blockManager.getBlock("core:RedLeaf"))
-                .setBarkType(blockManager.getBlock("core:OakTrunk")).setGenerationProbability(0.05f);
-
-        // Cactus
-        TreeGenerator cactus = new TreeGeneratorCactus().setTrunkType(blockManager.getBlock("core:Cactus")).setGenerationProbability(0.05f);
-
-        // Add the trees to the generator lists
-        treeGeneratorLookup.put(CoreBiome.MOUNTAINS, oakTree);
-        treeGeneratorLookup.put(CoreBiome.MOUNTAINS, pineTree);
-
-        treeGeneratorLookup.put(CoreBiome.FOREST, oakTree);
-        treeGeneratorLookup.put(CoreBiome.FOREST, pineTree);
-        treeGeneratorLookup.put(CoreBiome.FOREST, oakVariationTree);
-
-        treeGeneratorLookup.put(CoreBiome.SNOW, birchTree);
-
-        treeGeneratorLookup.put(CoreBiome.PLAINS, redTree);
-        treeGeneratorLookup.put(CoreBiome.PLAINS, oakTree);
-
-        treeGeneratorLookup.put(CoreBiome.DESERT, cactus);
+        blockManager = CoreRegistry.get(BlockManager.class);
     }
 
     @Override
     public void generateChunk(CoreChunk chunk, Region chunkRegion) {
         TreeFacet facet = chunkRegion.getFacet(TreeFacet.class);
-        BiomeFacet biomeFacet = chunkRegion.getFacet(BiomeFacet.class);
 
-        for (Map.Entry<Vector3i, Number> entry : facet.getRelativeEntries().entrySet()) {
-            float facetValue = entry.getValue().floatValue();
-            if (facetValue > 0) {
-                Vector3i pos = entry.getKey();
-                CoreBiome biome = biomeFacet.get(pos.x, pos.z);
-                for (TreeGenerator generator : treeGeneratorLookup.get(biome)) {
-                    if (generator.getGenerationProbability() > (facetValue / 256f)) {
-                        generator.generate(chunk, new FastRandom((long) facetValue), pos.x, pos.y, pos.z);
-                        break;
-                    }
-                }
-            }
+        for (Map.Entry<Vector3i, TreeGenerator> entry : facet.getRelativeEntries().entrySet()) {
+            Vector3i pos = entry.getKey();
+            TreeGenerator treeGen = entry.getValue();
+            int seed = relativeToWorld(facet, pos).hashCode();
+            Random random = new FastRandom(seed);
```

Comment:
It looks similar, but the seed value is generated per world location and independent from the iteration order and chunk/facet. At least unless I'm overlooking something here ..